### PR TITLE
Fix handling of missing 'select' clause

### LINKED ISSUE #2888: SA1103CodeFixProvider crash
StyleCop.Analyzers 1.1.1-rc.94, VS2017

```cs
 private object Resolve(object database)
 {
       var facilityRef = from member in database.Fac_MEMBERS
                         where member.;
}
```

I place cursor right afrer `Fac_MEMBERS` and call Quick Actions.
After this VS shows warning 

image

```
> System.ArgumentException : An item with the same key has already been added.
   at System.ThrowHelper.ThrowArgumentException(ExceptionResource resource)
   at System.Collections.Generic.Dictionary`2.Insert(TKey key,TValue value,Boolean add)
   at System.Collections.Generic.Dictionary`2.Add(TKey key,TValue value)
   at async StyleCop.Analyzers.ReadabilityRules.SA1103CodeFixProvider.GetTransformedDocumentFromSingleLineAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.CodeActions.CodeAction.GetChangedSolutionAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.CodeActions.CodeAction.ComputeOperationsAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.CodeActions.CodeAction.GetPreviewOperationsAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.Editor.Implementation.Suggestions.SuggestedAction.GetPreviewResultAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.Editor.Implementation.Suggestions.SuggestedActionWithNestedFlavors.<>c__DisplayClass11_0.<GetPreviewAsync>b__0(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.Extensions.IExtensionManagerExtensions.PerformFunctionAsync[T](<Unknown Parameters>)
```

### PR DESCRIPTION
**Pull request by sharwell**
Fixes #2888

### CODE REVIEW

File: StyleCop.Analyzers/StyleCop.Analyzers.Test/ReadabilityRules/SA1103UnitTests.cs [hunk 1]
```diff
@@ -365,5 +365,97 @@ public void TestMethod()
 
             await VerifyCSharpFixAsync(testCode, expectedDiagnostics, fixedTestCode, CancellationToken.None).ConfigureAwait(false);
         }
+
+        /// <summary>
+        /// Verifies that a query expression with directive trivia produces the expected results.
```
Comment (vweijsters) [inline StyleCop.Analyzers/StyleCop.Analyzers.Test/ReadabilityRules/SA1103UnitTests.cs]: 💡 Incorrect comment

File: StyleCop.Analyzers/StyleCop.Analyzers.Test/ReadabilityRules/SA1103UnitTests.cs [hunk 2]
```diff
@@ -365,5 +365,97 @@ public void TestMethod()
...
+    }
+}
+";
+
+            var expectedDiagnostics = new DiagnosticResult("CS0742", DiagnosticSeverity.Error).WithLocation(12, 38).WithMessage("A query body must end with a select clause or a group clause");
+            await VerifyCSharpFixAsync(testCode, expectedDiagnostics, testCode, CancellationToken.None).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Verifies that a query expression with directive trivia produces the expected results.
```
Comment (vweijsters) [inline StyleCop.Analyzers/StyleCop.Analyzers.Test/ReadabilityRules/SA1103UnitTests.cs]: 💡 Incorrect comment

File: StyleCop.Analyzers/StyleCop.Analyzers.Test/ReadabilityRules/SA1103UnitTests.cs (hunk 1 above)
Comment (sharwell) [reply StyleCop.Analyzers/StyleCop.Analyzers.Test/ReadabilityRules/SA1103UnitTests.cs]: Fixed, thanks!

File: StyleCop.Analyzers/StyleCop.Analyzers.Test/ReadabilityRules/SA1103UnitTests.cs (hunk 2 above)
Comment (sharwell) [reply StyleCop.Analyzers/StyleCop.Analyzers.Test/ReadabilityRules/SA1103UnitTests.cs]: Fixed, thanks!

File: StyleCop.Analyzers/StyleCop.Analyzers.Test/ReadabilityRules/SA1103UnitTests.cs [hunk 3]
```diff
@@ -365,5 +365,91 @@ public void TestMethod()
...
+                    Sources = { testCode },
+                    ExpectedDiagnostics =
+                    {
+                        Diagnostic(SA110xQueryClauses.SA1103Descriptor).WithLocation(11, 21),
+                        new DiagnosticResult("CS0742", DiagnosticSeverity.Error).WithLocation(12, 38).WithMessage("A query body must end with a select clause or a group clause"),
+                    },
+                },
+                FixedState =
+                {
+                    Sources = {fixedTestCode },
```
Comment (vweijsters) [inline StyleCop.Analyzers/StyleCop.Analyzers.Test/ReadabilityRules/SA1103UnitTests.cs]: Should be `Sources = { fixedTestCode }`, build failed on this.

Comment (sharwell) [reply StyleCop.Analyzers/StyleCop.Analyzers.Test/ReadabilityRules/SA1103UnitTests.cs]: Fixed

**Pull request merged**